lib/blkdb.c

```diff
@@ -20,6 +20,13 @@
 #include <ccoin/cstr.h>
 #include <ccoin/compat.h>		/* for fdatasync */
 
+#if defined(__APPLE__)
+static int fdatasync(int fd)
+{
+	return fsync(fd);
+}
+#endif
+
```

Comment:
I didn't notice the version in compat.h. But yes, the check there doesn't work.  Presumably because fdatasync does actually exist on mac but has a different signature, so configure manages to find it (despite the compile warnings) and sets HAVE_FDATASYNC.

I'll re-do the patch.  Thanks.


I'll fix it to use the compat.h version.